packages/fiber/src/core/utils.ts

```diff
@@ -282,18 +282,15 @@ export function applyProps(instance: Instance, data: InstanceProps | DiffSet) {
     // with their respective constructor/set arguments
     // For removed props, try to set default values, if possible
     if (value === DEFAULT + 'remove') {
-      if (targetProp && targetProp.constructor) {
-        // use the prop constructor to find the default it should be
-        value = new targetProp.constructor(...(memoized.args ?? []))
-      } else if (currentInstance.constructor) {
+      if (currentInstance.constructor) {
         // create a blank slate of the instance and copy the particular parameter.
         // @ts-ignore
         const defaultClassCall = new currentInstance.constructor(...(currentInstance.__r3f.memoizedProps.args ?? []))
-        value = defaultClassCall[targetProp]
-        // destory the instance
+        value = defaultClassCall[key]
+        // destroy the instance
```

Comment:
`key` only works for one level, so this won't work for pierced props. We'll have to rewrite this like v9 for proper resolution in all cases. I'll see if we can do a quick traversal here in the meantime.